### PR TITLE
Add example for anonymous sessions

### PR DESCRIPTION
**Pull request by mbarton**
## Changes 💡

Adds example for implementing anonymous sessions using a dummy CredentialsProvider as suggested in https://github.com/nextauthjs/next-auth/pull/3413#issuecomment-990052278.

### CODE REVIEW

File: docs/tutorials/anonymous-sessions.md [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+export default NextAuth({
+  // other options as above
+  session: ({ session, token }) => ({
+    ...session,
+    isLoggedIn: token.sub !== undefined
+  })
+});
+```
+
+Then in the frontend, you can sign in to the dummy provider without a redirect if not already logged in.
```
Comment (mbarton) [inline docs/tutorials/anonymous-sessions.md:114]: I've only added a f/e React example for now. I'm pretty sure signIn followed by getSession will work in getServerSideProps too but haven't had a chance to test. Thinking of raising a follow up PR if I do get a chance to.

**Comment by benjamintd** [comment]
Thanks for this example. This is exactly what I needed for a different use case - logging in anonymous users into a demo account in a specific context.

**Comment by luisrudge** [comment]
Hi, @mbarton thanks for the docs. I think they'd be a great addition!
Have you implemented this with Commerce Tools?

**Comment by luisrudge** [comment]
@mbarton yeah, I did something like this in the end. I do have a question, though: why are you calling `anonymousFlow` in the jwt callback? wouldn't that create a new session everytime a user asks for a jwt?

**Comment by rumeetgoradia** [comment]
This is great! Thanks for this.

One thing I'm wondering about is how to connect the anonymous account's data with the user's account after login. In my use case, for example, all visitors, anonymous or signed-in, will be persisted to a database. Visitors can perform different actions, creating different records in the database associated with their specific User record in the database. So, when an anonymous visitor decides to sign in (i.e. activates `signIn(<non-anon provider>)`), how can we create a callback such that everything in the database associated with the anonymous account gets transferred to the signed-in account?

I was thinking maybe do some magic in the `jwt` callback, but I don't think at that point we would have the anonymous account's ID. `signIn()` with an OAuth provider will always redirect, so by the time the user is signed in, the anonymous account ID is again unavailable. 

Any ideas?

**Pull request closed without merging**